### PR TITLE
feat: nullObject pattern

### PR DESCRIPTION
**Pull request by guipasmoi**

### CODE REVIEW

File: app/dataDefinitions/defects.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+const Defect = Symbol("Defect");
+
+export const isValid = object => object[Defect] === undefined;
+
+export const isFaulty = object => !isValid(object);
+
+export const getDefect = object => isValid(object) ? "" : object[Defect];
+
+export const NotFoundEntity = { [Defect]: "Not Found" };
+export const InvalidEntity = { [Defect]: "Invalid" };
```
Comment (guipasmoi) [inline app/dataDefinitions/defects.js:10]: add 
export const LoadingData= { [Defect]: "Loading" }

File: app/dataDefinitions/defects.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+const Defect = Symbol("Defect");
+
+export const isValid = object => object[Defect] === undefined;
+
```
Comment (guipasmoi) [inline app/dataDefinitions/defects.js:4]: remove jump line

File: app/dataDefinitions/defects.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+export const isFaulty = object => !isValid(object);
+
```
Comment (guipasmoi) [inline app/dataDefinitions/defects.js:6]: same

File: app/dataDefinitions/defects.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+export const getDefect = object => isValid(object) ? "" : object[Defect];
+
+export const NotFoundEntity = { [Defect]: "Not Found" };
```
Comment (guipasmoi) [inline app/dataDefinitions/defects.js:9]: may be we shouldn't use entity for naming but data or object or .... i don t know something

File: app/routes/screens/details/Details.js [hunk 5]
```diff
@@ -18,6 +24,8 @@ class Details extends Component {
 
   render() {
     const { athlete } = this.props;
+    if (isFaulty(athlete))
```
Comment (guipasmoi) [inline app/routes/screens/details/Details.js:27]: naming isFaulty? or may be isNotValid or .....?

File: package.json [hunk 6]
```diff
@@ -11,7 +11,6 @@
     "test": "jest",
     "lint": "eslint --quiet app",
     "devtools": "react-devtools",
-    "precommit": "lint-staged",
```
Comment (guipasmoi) [inline package.json]: yeah that bad

Comment (totorototo) [inline package.json]: put it back!

File: app/routes/screens/details/Details.js [hunk 7]
```diff
@@ -33,15 +41,13 @@ class Details extends Component {
   }
 }
 
-const getAthlete = (state, id) => {
-  if (id !== undefined && id !== 0 && state.entities.athletes) {
-    return state.entities.athletes[id];
-  }
-  return { firstname: "", lastname: "", profile: "" };
-};
+const getAthlete = (state, id) => getEntity(state, "athletes", id);
```
Comment (guipasmoi) [inline app/routes/screens/details/Details.js:44]: we may use curried version of getEntity 
id est : 
```
getAtheltes = getEntity("athletes")
getAtheltes(state,id)
```
may be

File: app/dataDefinitions/defects.js [hunk 8]
```diff
@@ -0,0 +1,10 @@
+const Defect = Symbol("Defect");
```
Comment (totorototo) [inline app/dataDefinitions/defects.js:1]: no more issue while using symbol? (rn context).

File: app/store.js [hunk 9]
```diff
@@ -1,5 +1,5 @@
 import Store from "./store/Store";
-import rootReducer from "./store/reducers/rootReducer";
+import rootReducer from "./store/state/rootReducer";
```
Comment (guipasmoi) [inline app/store.js:2]: i renamed reducers folder to state
i wish to have selectors and ... inside their scope

File: app/store/state/appState/appState.js [hunk 10]
```diff
@@ -12,7 +12,7 @@ export default combineReducers({
 //
 // const AppState = {
 //   "@@/data": {
-//     currentUserID: 123
+//     getCurrentUserID: 123
```
Comment (guipasmoi) [inline app/store/state/appState/appState.js:15]: wasn't intended

File: app/store/state/entities/selectors.js [hunk 11]
```diff
@@ -0,0 +1,10 @@
+import { NotFoundEntity, isValid } from "../../../dataDefinitions/defects";
+
+export const getEntity = (state, entityType, id) =>
+  state[entityType][id] === undefined ? NotFoundEntity : state[entityType][id];
+
+export const getEntities = (state, entityType, idList) =>
+  idList.map(id => getEntities(state, entityType, id));
+
+export const getValidEntities = (state, entityType, idList) =>
+  getEntities(state, entityType, idList).filter(isValid);
```
Comment (guipasmoi) [inline app/store/state/entities/selectors.js:10]: we can add more selectors but here is a few to show what we can do

File: app/dataDefinitions/defects.js (hunk 8 above)
Comment (guipasmoi) [reply app/dataDefinitions/defects.js:1]: may be. I don't know

Comment (guipasmoi) [reply app/dataDefinitions/defects.js:1]: we could use Object.defineProperty too may be

File: package.json (hunk 6 above)
Comment (guipasmoi) [reply package.json]: https://www.youtube.com/watch?v=FJaKjcxaJMs

**Pull request merged**